libs/utils/web3Websocket.ts

```diff
@@ -0,0 +1,117 @@
+import Web3 from "web3";
+import { WebsocketProvider } from "web3-core";
+import logger from "./logger";
+
+// CHECK WEB3_PROVIDER_OPTIONS
+const WEB3_PROVIDER_OPTIONS = {
+  clientConfig: {
+    keepalive: true,
+    keepaliveInterval: 10,
+  },
+  timeout: 3600000, // 1h in ms
+  reconnect: {
+    auto: true,
+    delay: 5000, // ms
+    maxAttempts: 120,
+    onTimeout: false,
+  },
+};
+
+// FYI see setEthConfig
+const TX_BLOCK_TIMEOUT = 2000;
```

Comment:
I agree with you 100%, however turns out this is not even about time :sweat_smile: from [web3 docs](https://web3js.readthedocs.io/en/v1.7.3/web3-eth.html#transactionblocktimeout): 

> The `transactionBlockTimeout` is used over socket-based connections. This option defines the amount of new blocks it should wait until the first confirmation happens, otherwise the PromiEvent rejects with a timeout error.